Introduction to Simulation.ipynb

```diff
@@ -56,59 +160,142 @@
+    "\n",
+    "# integrate the motion of the ball through time\n",
     "for i in range (0, timesteps-1):\n",
     "        \n",
     "        time_array[i+1] = time_array[i] + dt\n",
+    "        \n",
+    "        # kinematic equations\n",
     "        velocity_array[i+1] = velocity_array[i] + (g*dt)\n",
     "        height_array[i+1] = height_array[i] + 0.5*(velocity_array[i]+velocity_array[i+1])*dt\n",
     "\n",
     "        #if we have reached the ground, zero out the position and velocity\n",
     "        if height_array[i+1] <= 0:\n",
     "            velocity_array[i+1] = 0\n",
     "            height_array[i+1] = 0\n",
+    "            # print to the screen the time we hit the ground\n",
+    "            if hit_the_ground == False:\n",
+    "                print(\"The ball has hit the ground.\")\n",
+    "                print(f\"Time: {time_array[i+1]}\")\n",
+    "                hit_the_ground = True\n",
+    "\n",
+    "            \n",
+    "\n",
+    "#plot the height as a function of time. \n",
+    "plt.plot(time_array, height_array, c='blue')\n",
+    "\n",
+    "plt.ylabel('height (m)')\n",
+    "plt.xlabel('time (s)')\n",
+    "plt.title('Simulation of a ball falling from rest')\n",
     "\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Instead of only plotting the height, let us create a 2 panel plot to show both height and velocity as a function of time.  This is easy to accomplish with matplotlib.\n",
+    "\n",
+    "More information about using subplots can be found here: https://matplotlib.org/3.5.2/gallery/subplots_axes_and_figures/subplots_demo.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
     "ax = plt.subplot(2,1,1)\n",
     "ax.plot(time_array, height_array, c='blue')\n",
     "\n",
+    "plt.title('Simulation of a ball falling from rest')\n",
     "plt.ylabel('height (m)')\n",
     "\n",
     "ax = plt.subplot(2,1,2)\n",
-    "ax.plot(time_array, velocity_array, c='red')\n",
+    "ax.plot(time_array, velocity_array, c='red', )\n",
     "plt.ylabel('velocity (m/s)')\n",
     "\n",
     "plt.xlabel('time (s)')\n",
     "\n",
     "plt.show()"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Exercise 1: Changing the input parameters\n",
+    "\n",
+    "- Modify the timestep (```dt```) in the code above. How  does the time it takes to hit the ground change if you increase or decrease the timestep by an order of magnitude?  \n",
+    "- Change the initial velocity (```velocity_i```) to 10 m/s; does the model predict the behavior you would expect?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Exercise 2: Making the model more complex \n",
```

Comment:
Is exercise two supposed to modify the box of code above this, below this, or make a full new one? Just might be useful to have a separate box of code for this exercise where you mark in comments exactly where they should be looking to add the velocity scaling.